src/ndarray.jl

```diff
@@ -988,116 +988,143 @@ object (:class:`NDArray`) is returned. Otherwise, a tuple containing all the out
 
 **autogen:EMBED:ndarray:EMBED:autogen**
 =#
-function _import_ndarray_functions(;gen_docs=false)
-  n_ref = Ref{MX_uint}(0)
-  h_ref = Ref{Ptr{MX_handle}}(0)
-  @mxcall(:MXListFunctions, (Ref{MX_uint}, Ref{Ptr{MX_handle}}), n_ref, h_ref)
 
-  n_funcs = n_ref[]
-  h_funcs = pointer_to_array(h_ref[], n_funcs)
+function _get_ndarray_functions()
+  n = Ref{MX_uint}(0)
+  handles = Ref{Ptr{MX_handle}}(0)
 
-  if gen_docs
-    docs = Dict{Base.Symbol, AbstractString}()
-  end
+  @mxcall(:MXListFunctions, (Ref{MX_uint}, Ref{Ptr{MX_handle}}), n, handles)
 
-  for i = 1:n_funcs
-    func_handle = h_funcs[i]
+  funcs = unsafe_wrap(Array, handles[], n[])
+  return funcs
+end
 
-    #----------------------------------------
-    # get function information (human readable)
-    ref_name = Ref{char_p}(0)
-    ref_desc = Ref{char_p}(0)
-    ref_narg = Ref{MX_uint}(0)
+const _function_cache = Dict{Symbol, MX_handle}()
+function _get_function(name :: Symbol)
+  if !haskey(_function_cache, name)
+    handle = Ref{MX_handle}(0)
 
-    ref_arg_names = Ref{char_pp}(0)
-    ref_arg_types = Ref{char_pp}(0)
-    ref_arg_descs = Ref{char_pp}(0)
+    @mxcall(:MXGetFunction, (Cstring, Ref{MX_handle}), name, handle)
+    _function_cache[name] = handle[]
+    return handle[]
+  else
+    return _function_cache[name]
+  end
+end
 
-    ref_ret_type  = Ref{char_p}(0)
+function _get_function_description(handle :: MX_handle)
+  # get function information (human readable)
+  ref_name = Ref{char_p}(0)
+  ref_desc = Ref{char_p}(0)
+  ref_narg = Ref{MX_uint}(0)
+
+  ref_arg_names = Ref{char_pp}(0)
+  ref_arg_types = Ref{char_pp}(0)
+  ref_arg_descs = Ref{char_pp}(0)
+
+  ref_ret_type  = Ref{char_p}(0)
+
+  @mxcall(:MXFuncGetInfo,
+         (MX_handle, Ref{char_p}, Ref{char_p}, Ref{MX_uint}, Ref{char_pp},
+          Ref{char_pp}, Ref{char_pp}, Ref{char_p}),
+          handle, ref_name, ref_desc, ref_narg, ref_arg_names,
+          ref_arg_types, ref_arg_descs, ref_ret_type)
+
+  name = Symbol(unsafe_wrap(String, ref_name[]))
+
+  desc = unsafe_wrap(String, ref_desc[]) * "\n\n"
+  desc *= _format_docstring(Int(ref_narg[]), ref_arg_names, ref_arg_types, ref_arg_descs)
```

Comment:
You might also need to modify `_format_docstring` a bit, as it is written for the ReST syntax.